sdks/java/core/src/main/java/org/apache/beam/sdk/options/PipelineOptionsFactory.java

```diff
@@ -1406,8 +1406,16 @@ public boolean apply(@Nonnull String input) {
                       klass, entry.getKey(), closestMatches));
           }
         }
-
+        // We need to pull the method directly from klass, to retain generic information.
         Method method = propertyNamesToGetters.get(entry.getKey());
+        String methodName = method.getName();
+        try {
+          method = klass.getMethod(methodName);
+        } catch (NoSuchMethodException e) {
+          // If the method is not found, fall back to just using the registered value.  This
```

Comment:
You want to get all the methods from the interfaces and not the beanClass which you can pass in from the `validate` method since it was constructed there.

Note that the methods on the `beanClass` already have all the methods deduplicated while the `allInterfaceMethods` from `validate` will have duplicate definitions (because the methods are from multiple interfaces).

Also note that the validation logic that checked that parameter types/return types were equivalent were based off of class and not the generic type which will need updating.